Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm

```diff
@@ -164,22 +164,8 @@ void XPCServiceEventHandler(xpc_connection_t peer)
 
     xpc_connection_set_target_queue(peer, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0));
     xpc_connection_set_event_handler(peer, ^(xpc_object_t event) {
-        xpc_type_t type = xpc_get_type(event);
-        if (type != XPC_TYPE_DICTIONARY) {
-            RELEASE_LOG_ERROR(IPC, "XPCServiceEventHandler: Received unexpected XPC event type: %{public}s", xpc_type_get_name(type));
-            if (type == XPC_TYPE_ERROR) {
-                if (event == XPC_ERROR_CONNECTION_INVALID || event == XPC_ERROR_TERMINATION_IMMINENT) {
-                    RELEASE_LOG_FAULT(IPC, "Exiting: Received XPC event type: %{public}s", event == XPC_ERROR_CONNECTION_INVALID ? "XPC_ERROR_CONNECTION_INVALID" : "XPC_ERROR_TERMINATION_IMMINENT");
-                    // FIXME: Handle this case more gracefully.
-                    [[NSRunLoop mainRunLoop] performBlock:^{
-                        exitProcess(EXIT_FAILURE);
-                    }];
-                }
-            }
-            return;
-        }
 
-        handleXPCExitMessage(event);
+        handleXPCExitAndErrorMessage(event);
```

Comment:
This seems not right. Before this patch, the event handler early returns when event type is not dictionary, so that the `xpc_dictionary_get_string` function below won't be invoked. After this patch, the early return is removed and the function is always called (except when process is terminated inside `handleXPCExitAndErrorMessage`).

Reply:
Ah, great catch, fixed in latest patch!

Thanks for reviewing!